### DISCUSSION TITLE
Input APIs?

### DISCUSSION THREAD

**Discussion by willcastillo** [LlmLaraHub/laralamma#47] [Ideas]
Hello,

Are there any plans to add input APIs? Basically, create collections and adding data to them?

All the best,
W

**Comment by alnutile** [comment]
Good question someone else just asked me is there a LaraLlama HTTP Client.

I think I will add a new "Source" eg the way you get data into your collection as just an API

![CleanShot 2024-09-18 at 16 07 24@2x](https://github.com/user-attachments/assets/e2786a89-a18e-4ea9-8cf9-45bf72dd9336)

Right now the "Webhook" source acts as that since you can send say Github Webhooks to the system and then use your Source prompt to parse the data.

But I could add one that says "API" that then would allow you to add data to the system using the prompt to define the format (I am trying to make the prompts drive things so users are able to do more with no code)

Output has an API already btw so you can already get to the data just about putting data in. 
![CleanShot 2024-09-18 at 16 09 47@2x](https://github.com/user-attachments/assets/726005ec-60ed-448f-a0c2-af43a43da609)

Oh lastly there is an API I did for a chrome extension but that is still in the works and it allows you to send data to a source as well.

https://github.com/LlmLaraHub/larallama/tree/main/tests/fixtures/chrome_extension

This describes the API but the extension is still being tested by me and so is the api.

**Comment by willcastillo** [comment]
Sounds interesting. The use case I'm after is actually creating collections on the fly and adding documents to them so not sure if a webhook would work or be a good approach for it but I won't be surprised if I'm wrong!
